### ISSUE TITLE
the different of run() and eval in tensorflow

### ISSUE THREAD

**Issue by Jason-py** [Jason-py/tensorflow#1]
```
accuracy = tf.reduce_mean(tf.cast(correct_prediction, "float"))
sess.run(tf.initialize_all_variables())

for i in range(200):
  batch = mnist.train.next_batch(50)
  if i%100 == 0:
    train_accuracy = accuracy.eval(feed_dict={x:batch[0], y_: batch[1], keep_prob: 1.0})
    print( "step %d, training accuracy %g"%(i, train_accuracy))
  train_step.run(feed_dict={x: batch[0], y_: batch[1], keep_prob: 0.5})

print ("test accuracy %g"%accuracy.eval(feed_dict={
    x: mnist.test.images, y_: mnist.test.labels, keep_prob: 1.0}))
```

**this is some code in the example of tensorflow course, I want to tell the different of run() and eval(), because it make me realize the good use of function which applied by tensorflow.** 

**train_step.run(): feed the samples and train the nn change the parameter at the same time
accuracy.eval(): feed the samples but don't train model just get the result via the model**

_Do you want to know how I get this, because we can run twice the last two lines code and look the result is same?_

> test accuracy 0.9635
> test accuracy 0.9635

_It's same~_
